ruleskit/ruleset.py

```diff
@@ -651,6 +666,80 @@ def series_to_rule(self, srule: "pd.Series") -> Rule:
         rule._condition = HyperrectangleCondition(**condition_index)
         return rule
 
+    # noinspection PyUnresolvedReferences
+    def calc_predictions(self, y: [np.ndarray, "pd.Series"]) -> "pd.Series":
+        """
+        Will compute the prediction of each rule in the ruleset
+
+        Parameters
+        ----------
+        y: [np.ndarray, pd.Series]
+          The targets on which to evaluate the rules predictions, and possibly other criteria. Must be a 1-D np.ndarray
+          or pd.Series.
+
+        Returns
+        -------
+        "pd.Series"
+            A pd.Series with the rules as index and the values being the predictions (most probable class in case of
+            classification).
+        """
+        try:
+            import pandas as pd
+        except ImportError:
+            raise ImportError("RuleSet's calc_predictions requires pandas. Please run\npip install pandas")
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            if len(self) == 0:
+                if self.rule_type == ClassificationRule:
+                    return pd.DataFrame()
+                else:
+                    return pd.Series()
+
+            if self.stacked_activations is not None:
+                if self.rule_type == ClassificationRule:
+                    class_probabilities = functions.class_probabilities(self.stacked_activations, y)
+                    maxs = class_probabilities.max()
+                    return class_probabilities[class_probabilities == maxs].apply(
+                        lambda x: x.dropna().sort_index().index[0]
+                    )
+                else:
+                    return functions.conditional_mean(self.stacked_activations, y)
+            else:
+                if self.rule_type == ClassificationRule:
+                    [functions.class_probabilities(r, y) for r in self]
+                    return pd.Series(index=[str(r.condition) for r in self], data=[r.prediction for r in self])
+                else:
+                    return pd.Series(
+                        [functions.conditional_mean(r, y) for r in self], index=[str(r.condition) for r in self]
+                    )
```

Comment:
if self.rule_type == ClassificationRule:
    [functions.class_probabilities(r, y) for r in self]
    data = [r.prediction for r in self]
else:
    data = [functions.conditional_mean(r, y) for r in self]
return pd.Series(data=data, [functions.conditional_mean(r, y) for r in self])